cpp/src/strings/find.cu

```diff
@@ -216,6 +216,68 @@ std::unique_ptr<column> contains_fn(strings_column_view const& strings,
   return results;
 }
 
+/**
+ * @brief Utility to return a bool column indicating the presence of
+ * a string targets[i] in strings[i].
+ *
+ * Null string entries return corresponding null output column entries.
+ *
+ * @tparam BoolFunction Return bool value given two strings.
+ *
+ * @param strings Column of strings to check for `targets[i]`.
+ * @param targets Column of strings to be checked in `strings[i]``.
+ * @param pfn Returns bool value if target is found in the given string.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @return New BOOL column.
+ */
+template <typename BoolFunction>
+std::unique_ptr<column> contains_fn(strings_column_view const& strings,
+                                    strings_column_view const& targets,
+                                    BoolFunction pfn,
+                                    rmm::mr::device_memory_resource* mr,
+                                    cudaStream_t stream)
+{
+  auto strings_count = strings.size();
+  if (strings_count == 0) return make_numeric_column(data_type{BOOL8}, 0);
+
+  auto targets_count = targets.size();
+  CUDF_EXPECTS(targets_count > 0, "Must include at least one search target");
+
+  auto targets_column = column_device_view::create(targets.parent(), stream);
+  auto d_targets      = *targets_column;
+  auto strings_column = column_device_view::create(strings.parent(), stream);
+  auto d_strings      = *strings_column;
+  // create output column
+  auto results      = make_numeric_column(data_type{BOOL8},
+                                     strings_count,
+                                     copy_bitmask(strings.parent(), stream, mr),
+                                     strings.null_count(),
+                                     stream,
+                                     mr);
+  auto results_view = results->mutable_view();
+  auto d_results    = results_view.data<bool>();
+  // set the bool values by evaluating the passed function
+  thrust::transform(
+    rmm::exec_policy(stream)->on(stream),
+    thrust::make_counting_iterator<size_type>(0),
+    thrust::make_counting_iterator<size_type>(strings_count),
+    d_results,
+    [d_strings, pfn, d_targets] __device__(size_type idx) {
+      // empty target string returns true
+      if (d_targets.is_valid(idx) && d_targets.element<string_view>(idx).length() == 0) {
+        return true;
+      } else if (!d_strings.is_null(idx) && !d_targets.is_null(idx)) {
+        return static_cast<bool>(
```

Comment:
Why `static_cast<bool>` the result of `pfn`? Alternatively `return bool{pfn(...)};` would emit a warning/error if `pfn` returns non-bool.